### PR TITLE
feat: SingleProject - Commerce

### PR DESCRIPTION
**Pull request by vatsashah45**

### CODE REVIEW

**Comment by eriklimakc** [comment]
Sample updated with latest template.

Known issues so far:

1. **WinAppSDK** and **Android** The app uses the file **products.json** to store the products info. This file lives in the project root directory, but this file is not found by the code when trying to load the products. It only worked for Windows Packaged when I moved the file to Assets. But it didn't work for Android nor Windows Unpackaged.

2. When running the app on **Desktop** no windows is launched. Nothing really happens.

3. On **WASM** the app is launched, but just a black screen is shown.

I know there are more issues related to loading data. I will post them later after I test it more.

File: reference/Commerce/Commerce/Commerce.csproj [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿<Project Sdk="Uno.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>
+		$(DotNetVersion)-browserwasm;
+		$(DotNetVersion)-android;
+		$(DotNetVersion)-ios;
+		$(DotNetVersion)-maccatalyst;
+		$(DotNetVersion)-windows10.0.19041;
+		$(DotNetVersion)-desktop;
+	</TargetFrameworks>
```
Comment (eriklimakc) [inline reference/Commerce/Commerce/Commerce.csproj:10]:
```suggestion
    <TargetFrameworks>
      net8.0-desktop;
      net8.0-android;
      net8.0-ios;
      net8.0-maccatalyst;
      net8.0-windows10.0.19041;
      net8.0-browserwasm;
    </TargetFrameworks>
```

File: reference/Commerce/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<Project>
+  <PropertyGroup>
+    <DotNetVersion>net8.0</DotNetVersion>
```
Comment (eriklimakc) [inline reference/Commerce/Directory.Build.props:3]:
```suggestion
```

**Review by eriklimakc — CHANGES_REQUESTED** [review]

**Review by agneszitte — COMMENTED** [review]
@vatsashah45, @eriklimakc 
Few comments:
- Not sure if we can have a way to avoid having a Commerce folder inside an another Commerce folder here
- Remove unnecessary files
- And finally, if you can add a proper ReadMe for the sample please
image

---
To have a structure similar to what we have for ToDo here, with a `src` folder with the source inside, a proper README.md with the details of the sample, and a `doc` folder for additional documentation or assets needed for the README for example.
image
(cc @LProulx86 as well so we can use the same structure for the others samples)

**Comment by eriklimakc** [comment]
We're currently facing this issue on this app - https://github.com/unoplatform/uno/issues/16430

**Comment by agneszitte** [comment]
> We're currently facing this issue on this app - [unoplatform/uno#16430](https://github.com/unoplatform/uno/issues/16430)

@eriklimakc, @vatsashah45 now that https://github.com/unoplatform/uno/issues/16430 is closed, are there any issues left to finish the migration, please?
(cc @nickrandolph)

**Comment by eriklimakc** [comment]
> @eriklimakc, @vatsashah45 now that [unoplatform/uno#16430](https://github.com/unoplatform/uno/issues/16430) is closed, are there any issues left to finish the migration, please? (cc @nickrandolph)

Yes, I don't see the build error anymore, so that's fixed. But I see a regression, the Uno.Extensions.Storage.FileStorage issue (https://github.com/unoplatform/uno-private/issues/408) is back only on wasm, and that prevents products from loading:

image

cc @agneszitte @jeromelaban @nickrandolph

**Comment by rajamatt** [comment]
With latest packages the warnings for everything related to `appsettings` still appear on Wasm and other platforms:
image

However, the `products.json` file does not appear in the warnings. In fact, all the products are loaded and all feeds with products are working as expected.

cc: @agneszitte

**Comment by rajamatt** [comment]
## Favorites
- [ ] All items are favorited by default but they don't appear in the Favorites feed, we have to unfavorite them and favorite them again for them to appear in the Favorites feed.

**Comment by rajamatt** [comment]
## ProductsPage (search page)
- [ ] The "Products found" count is hardcoded

https://github.com/unoplatform/Uno.Samples/blob/bc7377700aae3c1721283b8ed8890bffda051903/reference/Commerce/src/Commerce/Views/ProductsPage.xaml#L66

- [ ] Searching for products only works if a filter is selected first
- [ ] Clicking on an item in the Search doesn't do anything (should it?)

**Comment by rajamatt** [comment]
## Miscellaneous
- [ ] About 1/5 of the time, navigating to the profile page causes a COMException (I think its because the profile pictures are random and one of them might be problematic but I'm not sure)

image

- Back buttons aren't visible but are functional (not sure if it's theming related or not)

image

- The left hand navbar's items could be spaced a bit better (they are very close together and the focus background stacks on adjacent items)

image

**Comment by kazo0** [comment]
@rajamatt can you create separate issues for those things you've noticed. And I can also reproduce them on current master so it doesnt seem to be related to this pr

**Comment by agneszitte** [comment]
Related issues: 
- https://github.com/unoplatform/Uno.Samples/issues/735

**Comment by rajamatt** [comment]
> @rajamatt can you create separate issues for those things you've noticed. And I can also reproduce them on current master so it doesnt seem to be related to this pr

Done

**Review by kazo0 — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

**Pull request merged**